sdk/tests/js/tests/tex-image-and-sub-image-utils.js

```diff
@@ -791,11 +791,71 @@ var TexImageUtils = (function() {
     return program;
   };
 
+  /**
+   * Return a list of unpack color spaces to test, supported by the specified
+   * WebGLRenderingContext.
+   */
+  var unpackColorSpacesToTest = function(gl)
+  {
+    if ('unpackColorSpace' in gl)
+      return ['srgb', 'display-p3'];
+    else
+      return [null];
+  }
+
+  /**
+   * For each entry in unpackColorSpaces, duplicate all of cases, adding an
+   * unpackColorSpace key with its value set to that entry to each case.
+   */
+  var crossProductTestCasesWithUnpackColorSpaces = function(cases, unpackColorSpaces)
+  {
+    var caseWithColorSpace = function(c, cs) { return {...c, ...{unpackColorSpace:cs}}; }
```

Comment:
Seeing the earlier discussion with @greggman , having done some reading, and not having seen this trick before - could you please add a comment like:

```
// Merge test case and unpackColorSpace using spread operator
```
? I spent some time wondering if two iterables with multiple items each were being folded into a dictionary.

Reply:
I want to respond constructively. Using the spread operator in 2022 is normal idiomatic JavaScript. Does it really need it described? I get that you didn't know it but now that you do if you browse most modern JavaScript you'll see it everywhere. For example [here's the react docs using it](https://reactjs.org/docs/higher-order-components.html)

If you really feel it's important then sure, keep the comment but then it seems like every use of the spread operator will need to be explained by a comment since you can't assume devs will find this one explanation. 

Is it just this particular one off usage?

Reply:
I'm certainly not helping the situation by using 1 and 2 character variable names almost exclusively :)

I've updated the variable names to be more self-documenting.